src/test/java/jp/co/yahoo/yosegi/message/formatter/json/TestJacksonMessageWriter.java

```diff
@@ -0,0 +1,288 @@
+    map.put( "key3" , "c" );
+    byte[] result = writer.create( map );
+    JsonNode node = mapper.readTree( result );
+    assertTrue( ( node instanceof ObjectNode ) );
+
+    ObjectNode objNode = (ObjectNode)node;
+    assertEquals( objNode.size() , 3 );
+    assertTrue( objNode.get("key1").isTextual() );
+    assertTrue( objNode.get("key2").isTextual() );
+    assertTrue( objNode.get("key3").isTextual() );
+    assertEquals( objNode.get("key1").asText() , "a" );
+    assertEquals( objNode.get("key2").asText() , "b" );
+    assertEquals( objNode.get("key3").asText() , "c" );
+  }
+
+  @Test
+  public void T_createMap_2() throws IOException {
+    ObjectMapper mapper = new ObjectMapper();
+    JacksonMessageWriter writer = new JacksonMessageWriter();
+    Map<Object,Object> map = null;
+    byte[] result = writer.create( map );
+    JsonNode node = mapper.readTree( result );
+    assertTrue( ( node instanceof ObjectNode ) );
+
+    ObjectNode objNode = (ObjectNode)node;
+    assertEquals( objNode.size() , 0 );
+  }
+
+  @Test
+  public void T_createParser_1() throws IOException {
+    ObjectMapper mapper = new ObjectMapper();
+    JacksonMessageWriter writer = new JacksonMessageWriter();
+    byte[] result = writer.create( new TestPrimitiveParser( false ) );
+    JsonNode node = mapper.readTree( result );
+    assertTrue( ( node instanceof ObjectNode ) );
+
+    ObjectNode arrayNode = (ObjectNode)node;
+    assertEquals( arrayNode.size() , 3 );
+    assertTrue( arrayNode.get("key1").isTextual() );
+    assertTrue( arrayNode.get("key2").isTextual() );
+    assertTrue( arrayNode.get("key3").isTextual() );
+    assertEquals( arrayNode.get("key1").asText() , "a" );
+    assertEquals( arrayNode.get("key2").asText() , "b" );
+    assertEquals( arrayNode.get("key3").asText() , "c" );
+  }
+
+  @Test
+  public void T_createParser_2() throws IOException {
+    ObjectMapper mapper = new ObjectMapper();
+    JacksonMessageWriter writer = new JacksonMessageWriter();
+    IParser parser = null;
+    byte[] result = writer.create( parser );
+    JsonNode node = mapper.readTree( result );
+    assertTrue( ( node instanceof NullNode ) );
+  }
+
+  @Test
+  public void T_createParser_3() throws IOException {
+    ObjectMapper mapper = new ObjectMapper();
+    JacksonMessageWriter writer = new JacksonMessageWriter();
+    byte[] result = writer.create( new TestPrimitiveParser( true ) );
+    JsonNode node = mapper.readTree( result );
+    assertTrue( ( node instanceof ArrayNode ) );
+    ArrayNode arrayNode = (ArrayNode)node;
+
+    assertEquals( arrayNode.size() , 3 );
+    assertTrue( arrayNode.get(0).isTextual() );
+    assertTrue( arrayNode.get(1).isTextual() );
+    assertTrue( arrayNode.get(2).isTextual() );
+    assertEquals( arrayNode.get(0).asText() , "a" );
+    assertEquals( arrayNode.get(1).asText() , "b" );
+    assertEquals( arrayNode.get(2).asText() , "c" );
+  }
+
+  @Test
+  public void T_createParser_4() throws IOException {
```

Comment:
Please write the test method name is to describe what is wanted to in the test instead of using number (it makes vague one).

Reply:
I changed the method name.